Sources/Sentry/SentryTraceContext.m

```diff
@@ -51,9 +51,16 @@ - (nullable instancetype)initWithTracer:(SentryTracer *)tracer
         return nil;
 
     NSString *userSegment;
-    if (scope.userObject.data[@"segment"] &&
-        [scope.userObject.data[@"segment"] isKindOfClass:[NSString class]])
-        userSegment = scope.userObject.data[@"segment"];
+
+    if (scope.userObject.segment) {
+        userSegment = scope.userObject.segment;
+    } else {
+        // For backwards compatibility
+        if (scope.userObject.data[@"segment"] &&
+            [scope.userObject.data[@"segment"] isKindOfClass:[NSString class]]) {
+            userSegment = scope.userObject.data[@"segment"];
+        }
```

Comment:
Since DS is still beta, I'd not do this, I'd rather consider a bug and fix it.

Reply:
That's great, thanks.